processor/attributesprocessor/testdata/config.yaml

```
@@ -294,7 +294,7 @@ processors:
       match_type: regexp
       attributes:
         # This attribute ('db.statement') must exist in the span and match the regex ('SELECT \* FROM USERS.*') for a match.
-        - {key: env, value: "'SELECT * FROM USERS WHERE ID=1'"}
+        - {key: env, value: "'SELECT \\* FROM USERS.*'"}
```

I think the comment is already in regex. The intended SQL command makes sense to me to be `SELECT * FROM USERS[optional WHERE/GROUP BY / ORDER BY]`. As such, I would "un-regex-ify" the comment and change it to the following :) 
```suggestion
        - {key: env, value: "'SELECT \* FROM USERS.*'"}
```

---

Hi Karen, I thought the same way as you do at first. But if we use one backslash to escaping, some tests won't pass. We need to add one more backslash somehow.

---

Hey Shibo, do you think it could be something like `"'SELECT \\\* FROM USERS.*'"` then? The current regex doesn't make sense to me. The first backslash would escape the second backslash, and the asterisk itself would not be escaped by anything. I tried with your regex using the `import re` library as well and it produces an error. Maybe this is not "real" regex though - I am not sure. Do you have the results of the failing test(s)?

---

Hey Shibo, hmm I think the problem is actually that character sequence in triple quotes was being considered as a raw string. I believe "SELECT '*' FROM USERS.*" should work. The contents within two set of quotes will be considered a raw string, while those in one set will be a string literal.

---

Hey Shibo, actually I think the failing tests you saw could've been because the character sequence in triple quotes was being considered as a raw string. I believe `"SELECT '*' FROM USERS.*"` should work though. The contents within one set of quotes will be considered a string literal, while those within two will be a raw string.

---

Hey Shibo, I think that was because the character sequence in triple quotes was being considered as a raw string:
```suggestion
        - {key: env, value: "SELECT '*' FROM USERS.*"}
```

---

Ah, I missed the syntax of double quotes and single quote. This makes sense. But why don't we just use a single quote then? i.e. `'SELECT * FROM USERS.*'`. What do you think?

---

I don't think that would work because as you can see from the `.*` at the end of `USERS.*`, the character `*` needs to be escaped. The triple quotes is so that the asterisk is considered a raw string instead of string literal.

---

Then `*` would be considered a string literal instead of a raw string - that would not work :)

---

(Well, it would pass the test, but not because it is registering the literal character `*` but because the `*` character is matching "any")

---

It seems that the tests in `attributesprocessor` package even don't test this value, you can change it to any value as you want and still pass all tests. The reason `"'SELECT \* FROM USERS.*'"` would cause a failure before is that this escape character is illegal, not the value itself.

---

Maybe we should add a test to validate the regex.

---

Also, I misunderstood the regex, so the regex is `/SELECT \* FROM USERS.*/g`. The original config value `'SELECT * FROM USERS WHERE ID=1'` is matched to this regex, actually. No need to make the change.

---

I will close this PR. Thanks for the cr Karen.

---

Hey, sorry I am a little late, but why was this PR closed? Seems the original issue still has not been fixed. 

The regex expression could be any syntactically valid expression, since there is no testing being done, as this file is a "sample configuration" for users to refer to (as mentioned in its [README](https://github.com/open-o11y/opentelemetry-collector/tree/d0d9b4b35441112cce35e245296a36e1d400746c/processor/attributesprocessor): `Refer to config.yaml for detailed examples on using the processor`).

So `/SELECT \* FROM USERS.*/g` would work but I would vouch for the suggested change still (which uses the double quotes)  since that's what it seems the rest of this file is doing for reg expressions (e.g. specifically in lines [169](https://github.com/open-o11y/opentelemetry-collector/blob/d0d9b4b35441112cce35e245296a36e1d400746c/processor/attributesprocessor/testdata/config.yaml#L169-L170), [273](https://github.com/open-o11y/opentelemetry-collector/blob/d0d9b4b35441112cce35e245296a36e1d400746c/processor/attributesprocessor/testdata/config.yaml#L273-L274), and [278](https://github.com/open-o11y/opentelemetry-collector/blob/d0d9b4b35441112cce35e245296a36e1d400746c/processor/attributesprocessor/testdata/config.yaml#L278-L279)).

Honestly I don't think it's a big deal whichever it is, as long as it's accurate :) The slash version regex seems fine too, and as it's your PR, it's comnpletely up to you :)

---

Hey, sorry I am a little late, but why was this PR closed? Seems the original issue still has not been fixed. 

As you mentioned, the regex expression in Line 297 could be any syntactically valid expression, since there is no testing being done. So `/SELECT \* FROM USERS.*/g` probably works? 

I would still vouch for the suggested change (using the double quotes)  since that's what it seems the rest of this file is doing for reg expressions (e.g. specifically in lines [169](https://github.com/open-o11y/opentelemetry-collector/blob/d0d9b4b35441112cce35e245296a36e1d400746c/processor/attributesprocessor/testdata/config.yaml#L169-L170), [273](https://github.com/open-o11y/opentelemetry-collector/blob/d0d9b4b35441112cce35e245296a36e1d400746c/processor/attributesprocessor/testdata/config.yaml#L273-L274), and [278](https://github.com/open-o11y/opentelemetry-collector/blob/d0d9b4b35441112cce35e245296a36e1d400746c/processor/attributesprocessor/testdata/config.yaml#L278-L279)).

Honestly I don't think it's a big deal whichever it is- the slash version regex seems fine, and as it's your PR, it's comnpletely up to you :) But I do think this PR should be reopened!

---

Oh, I think our understanding of the issue is different. 

The regex expression in Line 297 could be any syntactically valid expression (like you mentioned), which is why the current `'SELECT * FROM USERS WHERE ID=1'` is "valid" too. I think the issue is asking to change this code statement, so it actually matches what the comment says it is doing. 

Assuming that is the case, I was suggesting as a replacement of the current code is `"SELECT '*' FROM USERS.*"`. And I believe your suggested `/SELECT \* FROM USERS.*/` should work too. I like the former though, because I find it more "conventional". Also, it follows the convention of the code around it, by using double quotes for reg expressions (e.g. lines [169](https://github.com/open-o11y/opentelemetry-collector/blob/d0d9b4b35441112cce35e245296a36e1d400746c/processor/attributesprocessor/testdata/config.yaml#L169-L170), [273](https://github.com/open-o11y/opentelemetry-collector/blob/d0d9b4b35441112cce35e245296a36e1d400746c/processor/attributesprocessor/testdata/config.yaml#L273-L274), and [278](https://github.com/open-o11y/opentelemetry-collector/blob/d0d9b4b35441112cce35e245296a36e1d400746c/processor/attributesprocessor/testdata/config.yaml#L278-L279)). 

If that matches your understanding as well, I think the PR can be reopened. What do you think?

---

Also in reponse to your comment about "add[ing] a test to validate the regex" - I think according to this file's [README](https://github.com/open-telemetry/opentelemetry-collector/blob/main/processor/attributesprocessor/README.md) (near the bottom), this is simply a sample configuration that users can reference when creating their own? As such, there shouldn't be any testing that's needed here.